python/sim/function.js

```diff
@@ -1,15 +1,16 @@
-const { Function: SimFunction } = require("@winglang/sdk/lib/target-sim/function.js");
+const { App } = require("@winglang/sdk/lib/core");
 
-module.exports.Function = class Function extends SimFunction {
-  constructor(
-    scope,
-    id,
-    inflight,
-    props = {},
-  ) {
-    super(scope, id, inflight, props);
-    for (let e in props.env) {
-      inflight.inner.service.addEnvironment(e, props.env[e]);
+module.exports.handleSimInflight = (inflight, props) => {
+  for (let e in props.env) {
+    inflight.inner.service.addEnvironment(e, props.env[e]);
+  }
+
+  if (!App.of(inflight).isTestEnvironment) {
+    for (let key of ["AWS_REGION", "AWS_DEFAULT_REGION", "AWS_ACCESS_KEY_ID", "AWS_SECRET_ACCESS_KEY"]) {
```

Comment:
Does it throw an error if an env variable is missing?
Is it possible to throw an error in that case here?